src/notation/view/notationviewinputcontroller.cpp

```diff
@@ -171,6 +171,27 @@ void NotationViewInputController::initZoom()
     currentNotation()->viewState()->setMatrixInited(true);
 }
 
+void NotationViewInputController::initCanvasPos()
+{
+    RectF notationContentRect = m_view->notationContentRect();
+    double totalScoreWidth = notationContentRect.width();
+    double totalScoreHeight = notationContentRect.height();
+
+    double curScaling = m_view->currentScaling();
+    double viewWidth = m_view->width() / curScaling;
+    double viewHeight = m_view->height() / curScaling;
+
+    bool centerHorizontally = totalScoreWidth < viewWidth;
+    bool centerVertically = totalScoreHeight < viewHeight;
```

Comment:
Edge case: 
if `totalScoreWidth < viewWidth - 2 * canvasMargin`, then the score would be centered, but this results in the margin being less than `canvasMargin`. So, you get this:
<img width="1551" alt="Scherm­afbeelding 2024-09-16 om 12 59 58" src="https://github.com/user-attachments/assets/f63eeed9-ccfb-4e80-ba9e-6cf1cf450f6f">
instead of 
<img width="1551" alt="Scherm­afbeelding 2024-09-16 om 13 00 34" src="https://github.com/user-attachments/assets/4dfe9524-e267-4173-baff-9db9c8ee757f">